Sources/Internal/GrpcClient.swift

```diff
@@ -200,17 +224,42 @@ actor GrpcClient: CustomStringConvertible {
       DataConnectLogger
         .debug("executeMutation() receives response: \(resultsString, privacy: .private).")
 
-      guard results.errors.isEmpty else {
-        throw DataConnectError
-          .operationExecutionFailed(messages: createErrorJson(errors: results.errors))
-      }
-
-      if let decodedResults = try codec.decode(result: results.data, asType: resultType) {
-        return OperationResult(data: decodedResults)
-      } else {
-        DataConnectLogger
-          .debug("executeMutation() response: \(resultsString, privacy: .private) decode failed.")
-        throw DataConnectError.decodeFailed
+      // lets decode partial errors. We need these whether we succeed or fail
+      var errorInfoList = createErrorInfoList(errors: results.errors)
+
+      // check if decode succeeds
+      do {
+        let decodedResults = try codec.decode(result: results.data, asType: resultType)
+
+        // even though decode succeeded, we may still have received partial errors
+        if !errorInfoList.isEmpty {
+          let failureResponse = OperationFailureResponse(
+            rawJsonData: resultsString, errors: errorInfoList,
+            data: decodedResults
+          )
+          throw DataConnectOperationError.executionFailed(
+            response: failureResponse
+          )
+        } else {
+          return OperationResult(data: decodedResults)
+        }
+
+      } catch {
+        // we failed to decode
```

Comment:
I believe a similar thing occurs here as well.

Reply:
Fixed this one too.